test/evergreen.yml

```diff
@@ -626,6 +626,7 @@ functions:
       script: |
         set -o verbose
         ${PREPARE_TEST_ENV}
+        sudo sysctl -w kernel.perf_event_paranoid=1
```

Comment:
I had an impression such system-level setting would persist on the host (at least before reboot) so that other tasks running on the same host would get affected. Better double-check with the Evergreen team.

Reply:
You can't use syscall `SYS_perf_event_open` without enabling this functionality. It requires elevated permissions. The alternative is calling it using sudo. Sudo would be okay if I could only turn it on for the new tasks, but that is difficult as the cppsuite test evergreen.yml definitions are complex. It then also creates file permission issues. The better option which is what we have here changes this setting so that any process can access that information. I thought given that we are running with a spawnhost that would be okay as it'll be destroyed very shortly. 

@lukech I'm not sure what you are asking. Yes it will persist until the machine is torn down and recreated. We could simply turn it off again after the test completes. Does that work?

Reply:
@luke-pearson that's fine, I have no issue with using the `sysctl` and agree that the transient spawn host makes it OK. I was asking more about the level of 1.

Reply:
@luke-pearson 

> @lukech I'm not sure what you are asking. Yes it will persist until the machine is torn down and recreated. We could simply turn it off again after the test completes. Does that work?

I believe Evergreen works by trying to keep the hosts up as long as they can to run tasks scheduled on the same distro until there's no task in the distro queue that some hosts will be torn down to save cost. That means once a new cpp perf task runs on the host, later tasks running on the same host would be 'affected' by this system setting. We don't know what value is set in the distro image for this specific parameter "kernel.perf_event_paranoid" (hence the suggestion of checking with the Evergreen team). You don't need to enable the setting in your test if it's already been enabled in the disk image. 

Probably a safer way is to check the current parameter value in your test before enabling it and change back to the original value (if needed) after your test. Again, I don't know if the Evergreen team has any best practices to handle such use cases.

Reply:
@johndaniels, would you be able to weigh in here? The perf event paranoia level is a bit unknown to me. Also @lukech has a question which you may be able to answer better than me.

Reply:
We _should_ already have this solved for the [existing benchmark](https://github.com/10gen/mongo/blob/0c60aebae00d5e10c1192ef467232fcc040a585e/src/mongo/db/service_entry_point_common_bm.cpp#L4) that uses these performance syscalls. We should probably investigate what they do. I can't find a call to update that setting in the mongo repo at least.

Reply:
@korteland I've been looking into this with rather curious results. On my own spawnhost a level of `3` is sufficient to run the test collecting cpu instruction counts. But the documentation makes no mention to levels greater than two:
```
perf_event_paranoid:

Controls use of the performance events system by unprivileged
users (without CAP_SYS_ADMIN).  The default value is 2.

 -1: Allow use of (almost) all events by all users
     Ignore mlock limit after perf_event_mlock_kb without CAP_IPC_LOCK
>=0: Disallow ftrace function tracepoint by users without CAP_SYS_ADMIN
     Disallow raw tracepoint access by users without CAP_SYS_ADMIN
>=1: Disallow CPU event access by users without CAP_SYS_ADMIN
>=2: Disallow kernel profiling by users without CAP_SYS_ADMIN
```
Interestingly the default on my host was `4`. I'll test this on a spawn host to confirm if the behaviour is similar there.

Reply:
Very strange - thanks for taking a look.

Reply:
On our perf hosts level 2 is the highest level that still runs the test. I've made a change that saves and restores the default level as required. 668fc7189

Reply:
@lukech, @korteland if this is satisfactory could you please close out this thread?

Reply:
Thanks - the changes here look good.

Reply:
Looking good, thanks.